### PR TITLE
Update DG for Upload Picture feature

### LINKED ISSUE #94: Create Diagrams for Upload Picture Feature for v1.3
Update Developers Guide
Create UML sequence and activity diagrams

### PR DESCRIPTION
**Pull request by zzthian**
Refactor ImageStorage to handle ALL picture processing.
Add some limited test cases for 
Create UML diagrams for DG.
Update DG to reflect implementation of UploadPictureCommand as well as its design considerations.
Closes #94

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -254,8 +255,51 @@ Figure No. Sequence diagram for AttendanceDeleteCommand
 
 **Aspect: Size of attendance lists**
 - Current Implementation: Max size of 12
+<<<<<<< HEAD
```
Comment (gowribhat) [inline docs/DeveloperGuide.md]: I think this needs to be removed

File: src/main/java/seedu/address/storage/ImageStorage.java [hunk 2]
```diff
@@ -79,6 +85,33 @@ private static String getImagePath(Student student) {
         ).toString();
     }
 
+    /**
+     * Returns whether the given file is a .jpg file.
+     * @param file
```
Comment (gowribhat) [inline src/main/java/seedu/address/storage/ImageStorage.java:91]: Need to leave a line before params by Javadoc conventions

**Review by gowribhat — COMMENTED** [review]
Minor issues but otherwise LGTM! 💯

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (zzthian) [reply docs/DeveloperGuide.md]: HAHAHA yikes yes u are right

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 3]
```diff
@@ -4,19 +4,19 @@ skinparam arrowThickness 1.1
...
-Person *--> Name
-Person *--> Phone
-Person *--> Email
-Person *--> ClassGroup
-Person *--> Address
+Student *--> Name
+Student *--> Phone
+Student *--> Email
+Student *--> ClassGroup
+Student *--> Address
```
Comment (thamruicong) [inline docs/diagrams/BetterModelClassDiagram.puml]: this one should update to studentId? to reflect the studentId refactoring

Comment (zzthian) [reply docs/diagrams/BetterModelClassDiagram.puml]: Yes I missed this. Thanks

**Review by leewaichong — COMMENTED** [review]
LGTM

**Review by thamruicong — APPROVED** [review]
LGTM

**Pull request merged**